api/v1beta1/azurecluster_webhook.go

```diff
@@ -128,6 +128,44 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	oldSubnetMap := make(map[string]SubnetSpec, len(old.Spec.NetworkSpec.Subnets))
+	oldSubnetIndex := make(map[string]int, len(old.Spec.NetworkSpec.Subnets))
+	for i, subnet := range old.Spec.NetworkSpec.Subnets {
+		oldSubnetMap[subnet.Name] = subnet
+		oldSubnetIndex[subnet.Name] = i
+	}
+	for i, subnet := range c.Spec.NetworkSpec.Subnets {
+		if oldSubnet, ok := oldSubnetMap[subnet.Name]; ok {
+			// Verify the CIDR blocks haven't changed for an owned Vnet.
+			// A non-owned Vnet's CIDR block can change based on what's
+			// defined in the spec vs what's been loaded from Azure directly.
+			if old.Spec.NetworkSpec.Vnet.Tags.HasOwned(old.Name) && !reflect.DeepEqual(subnet.CIDRBlocks, oldSubnet.CIDRBlocks) {
```

Comment:
This technically allows the cidr block to be modified in the brief moments before the Vnet is created (because the tags haven't been set yet) but once the Vnet has been created it becomes immutable. That seems acceptable to me since it doesn't create any changes in the subnet since it doesn't exist yet. Thoughts?